### ISSUE TITLE
Create a box-plot annotation

### ISSUE THREAD

**Issue by lpatiny** [zakodium-oss/react-plot#406]
![image](https://user-images.githubusercontent.com/1484241/168286048-be0f5580-f2db-4b84-959a-2b691dd871ec.png)

q1
median
q3
min
max

**Comment by lpatiny** [comment]
![image](https://user-images.githubusercontent.com/1484241/168292117-4b051ca6-558b-47ba-94f8-90d70ecca21b.png)

We have identified couple of properties that could be used. Some of them are in our 'units' system, pixel, axis value or percent:
- width (the width of the box plot on the yAxis
- y the position on the yAxis
- min, max, q1, q3 and median, the values on the xAxis

We also identified various properties to change the style of the box plot

Please take care  that we can invert xAxis and yAxis !!! This means that if we want to draw a vertical box plot we should 'simply' add
- xAxis: 'y'
- yAxis: 'x'

Default values could be no fill, black color, 1 px width except for median that would be 2 px wide.

![image](https://user-images.githubusercontent.com/1484241/168294539-4d94b9a9-6a8c-4909-ba1e-e7ca9cbb837e.png)

@wadjih-bencheikh18 could you try to implement this and of course tell us if we forgot something or if it seems weird.

One of the example in storybook could be something like

![image](https://user-images.githubusercontent.com/1484241/168296003-3d1a26d1-fdb3-4bc8-bb66-5ab4019ff829.png)

In which we display horizontal axis and not the vertical axis. But we should also have an horizontal axis from 0 to 1000 but the box plot that has the values 200,300,400,500,700.

**Comment by stropitek** [comment]
> * width (the width of the box plot on the yAxis

This seems confusing to me since width is usually related to the horizontal axis. `minMaxWidth` being an horizontal property but `width` being a vertical property seems confusing to me.

Boxplots are most often vertical from what I've seen in publications so maybe that should be the behavior with `xAxis=x` and `yAxis=y`. Then we could keep `width` but `minMaxWidth` -> `minMaxHeight`, etc...

**Comment by stropitek** [comment]
Other possibility might be to rename "width" to "size" or "breadth"? I personally prefer "breadth" to "width"

**Comment by targos** [comment]
Not sure about the main width/breadth, but I think we should keep minMaxWidth, as it refers to the stroke width, not a particular dimension

**Comment by wadjih-bencheikh18** [comment]
> minMaxWidth

it's minMaxStorkeWidth not the length of the line check https://github.com/zakodium-oss/react-plot/pull/408

**Comment by wadjih-bencheikh18** [comment]
is it supposed to be symmetric , maybe the user want to make max line diffrent then min line

**Comment by targos** [comment]
> we can add an additional props : style : for the rect style minMaxStyle : for minMax line style whiskerStyle medianStyle

yes, go for it!

**Comment by stropitek** [comment]
> but I think we should keep minMaxWidth

true, it was maybe a bit ambiguous so `minMaxStyle: {strokeWidth: 2}` would resolve that ambiguity

> style : for the rect style

I guess we could call it `boxStyle` as well

**Comment by targos** [comment]
What about we remove all `xxxWidth` props (and `xxxStyle` with `strokeWidth` must be used) and only keep `xxxStyle` and `xxxColor` ?

**Comment by wadjih-bencheikh18** [comment]
style props added
